src/crypto/index.js

```diff
@@ -281,42 +315,29 @@ function _maybeUploadOneTimeKeys(crypto) {
 
         // Ask olm to generate new one time keys, then upload them to synapse.
         crypto._olmDevice.generateOneTimeKeys(keysThisLoop);
-        return _uploadOneTimeKeys(crypto).then(() => {
+        return _uploadOneTimeKeys(crypto).then((res) => {
+            const keyCount = res.one_time_key_counts.signed_curve25519 || 0;
```

Comment:
what's going on here? what is the purpose of || 0? why are we suddenly ignoring `numberToGenerate` but *only* in this case?

(I think I kinda see what you're doing, but wouldn't passing the current `keyCount `into `uploadLoop` instead of `numberToGenerate` be a rather clearer way of doing it?)